active/0000-move-rename.md

```diff
@@ -0,0 +1,175 @@
+
+The arguments; blueprint, source, and dest should be strings with no extension, as it is derived from the files associated with the defined blueprint argument.
+
+#### Arguments
+| name | description |
+| ---- | ----------- |
+| blueprint | name of the blueprint to use as context |
+| source | name of existing generated blueprint |
+| dest | path to generate destination |
+
+#### Options
+| name | description |
+| ---- | ----------- |
+| dry-run | run and log output without actually executing |
+| verbose | log all actions |
+| force | overwrite any existing destination files |
+| structure | structure to use as context for looking up path (pod, type, addon) |
+
+### Examples:
+Given the following app structure:
+```
+app
+├── components
+│   └── foo-bar.js
+│   └── bar
+│       └── foo-baz.js
+├── routes
+├── templates
+│   └── components
+│       └── foo-bar.js
+│       └── bar
+│           └── foo-baz.js
+...
+```
+```
+ember r component foo-bar bar-foo
+```
+
+### Process
+
+1. `ember r <blueprint> <source> <dest> [options]...`
+2. pre-process:
+  1. generate paths
+    * lookup BLUEPRINT
+    * get fileInfos for any files that would be generated
+    * construct paths for source and dest
+  2. pre-verify:
+    * SOURCE path exists
+    * DEST path does not already exist
+  4. beforeMove hook
+  5. prompt user
+3. process:
+  1. create dest dir(s) they don't already exist
+  2. loop through file list and run `ember mv <source> <dest> <options>`
+4. post-process:
+  1. afterMove hook
+
+# Out of scope
+
+Renaming entire directories with `ember mv` would be a nice to have, specifically for the capability to update path references in a project. It will likely need to wait for a simpler, solid version of the command.
+
+To reduce scope we will likely require the project to be versioned and only use `git mv`. Instead of falling back to `mv` or `move`, we will throw a warning that git is required for the command. Eventually a fallback can be provided, but for the initial pass it will not.
+
+# Drawbacks
+
+Because the mv command has a long established history, the limited functionality of this implementation could cause confusion for many users. There also may be many corner cases that won't immediately be covered by the command, possibly breaking apps, causing more issues than it solves. As long as the command leaves a trail and log of what it did, undoing any accidental changes should be managable.
+
+If the mv command is used to update several files in sequence, it will not be very efficient, as it will be repeating the same task several times without the ability to consolidate. However, performance is less of a concern in regard to migration/refactoring, as the user shouldn't need to do so frequently.
+
+# Alternatives
+
+Ember-cli alternatives are unknown, the only way to do this currently is manually through the command line and with a text editor with search and replace capabilities.
+
+# Unresolved questions
+
```

Comment:
* does this rename components in `HBS` files?
* does this rename/move route information in config/router.js?